Backpack/src/main/java/net/skyscanner/backpack/skeleton/BpkCircleSkeleton.kt

```diff
@@ -0,0 +1,95 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.skeleton
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.View
+import androidx.annotation.AttrRes
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.use
+
+class BpkCircleSkeleton @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  @AttrRes defStyleAttr: Int = 0,
+) : View(context, attrs, defStyleAttr) {
+  val paint = Paint().apply {
+    color = context.getColor(R.color.__skeletonBackground)
+    style = Paint.Style.FILL
+  }
+
+  enum class CircleSize(
+    internal val id: Int,
+  ) {
+    None(0),
```

Comment:
having none here feels a little wrong as in that case you wouldn't use a skeleton. maybe we could have a sensible default of either small or large and a third custom option which would then make use of the diameter? this would also remove any inconsistencies in call order between circlesize and diameter

Reply:
make sense. Will change "None" to "CustomSize".